src/backend/distributed/commands/role.c

```diff
@@ -893,10 +892,31 @@ GenerateGrantRoleStmtsOfRole(Oid roleid)
 		grantRoleStmt->grantor = NULL;
 
 #if PG_VERSION_NUM >= PG_VERSION_16
+
+		/* inherit option is always included */
+		DefElem *inherit_opt;
+		if (membership->inherit_option)
+		{
+			inherit_opt = makeDefElem("inherit", (Node *) makeBoolean(true), -1);
+		}
+		else
+		{
+			inherit_opt = makeDefElem("inherit", (Node *) makeBoolean(false), -1);
+		}
+		grantRoleStmt->opt = list_make1(inherit_opt);
+
+		/* admin option is false by default, only include true case */
 		if (membership->admin_option)
 		{
-			DefElem *opt = makeDefElem("admin", (Node *) makeBoolean(true), -1);
-			grantRoleStmt->opt = list_make1(opt);
+			DefElem *admin_opt = makeDefElem("admin", (Node *) makeBoolean(true), -1);
+			grantRoleStmt->opt = lappend(grantRoleStmt->opt, admin_opt);
+		}
+
+		/* set option is true by default, only include false case */
+		if (!membership->set_option)
+		{
+			DefElem *set_opt = makeDefElem("set", (Node *) makeBoolean(false), -1);
+			grantRoleStmt->opt = lappend(grantRoleStmt->opt, set_opt);
```

Comment:
To make Codecov happy here, we need to add tests for changes in role.c. So, you are doing these changes in `role.c` to be able to propagate proper membership options for a role when adding a new node to the Citus cluster. Roles are automatically distributed in a Citus cluster, so when adding a new node, the role along with all its options should be propagated to the new worker node.

You can find such tests under `create_role_propagation.sql`. It currently only has tests for the admin option, for example:
https://github.com/citusdata/citus/blob/main/src/test/regress/sql/create_role_propagation.sql#L39-L58
The example I linked above runs a grant command with admin option, then adds a new worker node (`:worker_2_port`), then goes to the new worker node to check the admin_option FROM pg_auth_members.


However, note that you cannot directly add tests to create_role_propagation test. These new tests that you will add with INHERIT and SET should be part of `pg16.sql` because they should only run with PG16.